src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -104,6 +108,11 @@ pub struct MergeIteratorInner<I: HummockIterator, NE: NodeExtraOrderInfo> {
     heap: BinaryHeap<Node<I, NE>>,
 
     last_table_key: Vec<u8>,
+
+    // compactor task progress report
+    task_progress: Option<Arc<TaskProgress>>,
```

Comment:
I think it is a bad idea to embed a compactor related concept into a generic merge iterator. cc @wenym1

Reply:
Also, `compact_and_build_sst` takes a `HummockIterator`, not `MergeIterator` so it is still possible we miss to update task progress. How about having a wrapper iterator that `impl HummockIterator` to wrap the underlying `HummockIterator` to do key progress tracking.